src/detect-ssh-proto.c

```diff
@@ -101,20 +100,22 @@ void DetectSshProtocolRegister(void)
     sigmatch_table[DETECT_AL_SSH_PROTOCOL].Setup = DetectSshProtocolSetup;
     sigmatch_table[DETECT_AL_SSH_PROTOCOL].flags |= SIGMATCH_INFO_STICKY_BUFFER | SIGMATCH_NOOPT;
 
-
-    DetectAppLayerMpmRegister2(BUFFER_NAME, SIG_FLAG_TOSERVER, 2,
-            PrefilterGenericMpmRegister, GetSshData,
-			ALPROTO_SSH, SshStateBannerDone),
-    DetectAppLayerMpmRegister2(BUFFER_NAME, SIG_FLAG_TOCLIENT, 2,
-            PrefilterGenericMpmRegister, GetSshData,
-			ALPROTO_SSH, SshStateBannerDone),
-
-    DetectAppLayerInspectEngineRegister2(BUFFER_NAME,
-            ALPROTO_SSH, SIG_FLAG_TOSERVER, SshStateBannerDone,
-            DetectEngineInspectBufferGeneric, GetSshData);
-    DetectAppLayerInspectEngineRegister2(BUFFER_NAME,
-            ALPROTO_SSH, SIG_FLAG_TOCLIENT, SshStateBannerDone,
-            DetectEngineInspectBufferGeneric, GetSshData);
+    sigmatch_table[DETECT_SSH_PROTOCOL].name = KEYWORD_NAME_LEGACY;
```

Comment:
instead of this new registration we should be able to just use `alias` as it is set in the original code